spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.prototypebean.dynamicarguments;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import junit.framework.Assert;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(loader = AnnotationConfigContextLoader.class, classes = EmployeeConfig.class)
+public class DynamicBeanUnitTest {
+
+    @Test
+    public void givenPrototypeBean_WhenFunction_ThenNewInstanceReturn() {
+
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(
+                EmployeeConfig.class);
```

Comment:
Let's use four-space indentation (from previous line) for the continuation lines like this one and a few others noted below.

Reply:
updated